### ISSUE TITLE
Add comments to document APIs

### ISSUE THREAD

**Issue by SimiHunjan** [hashgraph/hedera-sdk-cpp#216]
### Problem

As a developer, I want to have access to API level documentation when building applications.

### Solution

Add comments to explain the supported APIs.

**Comment by rwalworth** [comment]
Hey @deyanzz, would you be able to look through all our APIs in the code base and make sure they're all documented? I took a look through and I believe we have everything documented, but another set of eyes taking a look would be good. If you can't find anything we can probably close this, does that sound good @SimiHunjan ?

**Comment by SimiHunjan** [comment]
sounds great to me thank you!

**Comment by deyanzz** [comment]
Today I passed through 50 header files from 104 in total, but unfortunately I didn't make it to the end with my double-checks. I committed all of my progress in branch `00216-add-comments-to-document-apis` and after a brief talk with @rwalworth we decided it could be better he to finish this task today. So I'm assigning this issue to him.
The last file I started to review and check for missing doc comments is `ChunkedTransaction.h`- I walked through all of the header files before it.

In some of the files I did the following:
 - params renaming;
 - remove some mentioning of exceptions;
 - use the default code formatting proposed by the installed Extensions;
 - improve formatting;
 - add some white spaces in comments;
